include/ibm/management_console_rest.hpp

```diff
@@ -533,14 +533,47 @@ inline void
             asyncResp->res.jsonValue["TransactionID"] = var;
             return;
         }
-        BMCWEB_LOG_DEBUG << "There is a conflict with the lock table";
-        asyncResp->res.result(boost::beast::http::status::conflict);
+
+        // Check if the session is active. If active, send back the lock
+        // conflict, else clear the stale entry in the lock table.
         auto var =
             std::get<std::pair<uint32_t, LockRequest>>(conflictStatus.second);
+        std::string sessionId = std::get<0>(var.second);
+        auto session =
+            persistent_data::SessionStore::getInstance().getSessionByUid(
+                sessionId);
+
+        if (session == nullptr)
+        {
+            BMCWEB_LOG_ERROR << "Releasing lock of the session id(stale): "
+                             << sessionId;
+            // clear the stale entry
+            crow::ibm_mc_lock::Lock::getInstance().releaseLock(sessionId);
+
+            // create the new lock record for the session
+            varAcquireLock =
+                crow::ibm_mc_lock::Lock::getInstance().acquireLock(t);
+            conflictStatus =
+                std::get<crow::ibm_mc_lock::Rc>(varAcquireLock.second);
+            if (!conflictStatus.first)
+            {
+                asyncResp->res.result(boost::beast::http::status::ok);
+
+                auto var = std::get<uint32_t>(conflictStatus.second);
+                nlohmann::json returnJson;
+                returnJson["id"] = var;
```

Comment:
why was this needed ?

Reply:
so the following actually would have been reported as a lock conflict earlier
Now what it does is, removes the invalid session stale entry then creates a lock record for the current session and returns the json


curl -k -H "X-Auth-Token:$bmc_token" -X POST - "Content-type: application/json" -d '{ "Request" :[ { "LockType":"Write", "SegmentFlags": [ {"LockFlag":"DontLock","SegmentLength":1}, {"LockFlag":"DontLock","SegmentLength":2}, {"LockFlag":"LockAll","SegmentLength":1}], "ResourceID": 288241371267989504 } ]}' https://${bmc}/ibm/v1/HMC/LockService/Actions/LockService.AcquireLock
{
  "TransactionID": 2
}

Reply:
no i was asking ,
```  nlohmann::json returnJson;
     returnJson["id"] = var;
```
why was this needed ?